airflow/providers/amazon/aws/hooks/elasticache_replication_group.py

```diff
@@ -0,0 +1,289 @@
+        self.initial_poke_interval = initial_poke_interval
+
+        super().__init__(client_type='elasticache', *args, **kwargs)
+
+    def create_replication_group(self, config):
+        """
+        Call ElastiCache API for creating a replication group
+
+        :param config: Configuration for creating the replication group
+        :type config: dict
+        :return: Response from ElastiCache create replication group API
+        :rtype: dict
+        """
+        return self.conn.create_replication_group(**config)
+
+    def delete_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for deleting a replication group
+
+        :param replication_group_id: ID of replication group to delete
+        :type replication_group_id: str
+        :return: Response from ElastiCache delete replication group API
+        :rtype: dict
+        """
+        return self.conn.delete_replication_group(ReplicationGroupId=replication_group_id)
+
+    def describe_replication_group(self, replication_group_id):
+        """
+        Call ElastiCache API for describing a replication group
+
+        :param replication_group_id: ID of replication group to describe
+        :type replication_group_id: str
+        :return: Response from ElastiCache describe replication group API
+        :rtype: dict
+        """
+        return self.conn.describe_replication_groups(ReplicationGroupId=replication_group_id)
+
+    def get_replication_group_status(self, replication_group_id):
+        """
+        Get current status of replication group
+
+        :param replication_group_id: ID of replication group to check for status
+        :type replication_group_id: str
+        :return: Current status of replication group
+        :rtype: str
+        """
+        return self.describe_replication_group(replication_group_id)['ReplicationGroups'][0]['Status']
+
+    def is_replication_group_available(self, replication_group_id):
+        """
+        Helper for checking if replication group is available or not
+
+        :param replication_group_id: ID of replication group to check for availability
+        :type replication_group_id: str
+        :return: True if available else False
+        :rtype: bool
+        """
+        return self.get_replication_group_status(replication_group_id) == 'available'
+
+    def _has_reached_terminal_state(self, replication_group_id):
+        """
+        Helper for checking if we should stop poking replication group for availability or not
+
+        :param replication_group_id: ID of replication group to check for terminal state
+        :type replication_group_id: str
+        :return: Flag to check if availability-check should be stopped or not and current status
+        :rtype: (bool, str)
+        """
+        status = self.get_replication_group_status(replication_group_id=replication_group_id)
+
+        return status in self.TERMINAL_STATES, status
```

Comment:
```suggestion
```
WDYT of removing this and being explicit about it (see line 146 & 147)

Reply:
Or pass status to this function and return only if it is a terminal state. Returning two values where one depends on the other one looks to me that this should be two statements like I suggested in line 146 & 147.

Reply:
I think it makes sense. Since it was only getting used at one place, we can write 2 statements (as you suggested) and remove that method which was returning multiple values.
Now that you pointed out, I think it was not correct because the method name was `_has_reached_terminal_state` but it was returning `status` also along with the bolen flag which I think was not correct.